clients/drcachesim/tests/scheduler_unit_tests.cpp

```diff
@@ -5676,13 +5732,14 @@ test_unscheduled_initially_rebalance()
     // We need the initial runqueue assignment to be unbalanced.
     // We achieve that by using input bindings.
     // This relies on knowing the scheduler takes the 1st binding if there
-    // are any: so we can set to all cores and these will all pile up on output #0
+    // are any if the bindings don't include all cores: so we can set to all-but-one
+    // core and these will all pile up on output #0
     // prior to the init-time rebalance.  That makes output
     // #0 big enough for a rebalance attempt, which causes scheduler init to fail
     // without the i#7318 fix as it can only move one of those blocked inputs and
     // so it hits an IDLE status on a later move attempt.
     std::set<scheduler_t::output_ordinal_t> cores;
-    cores.insert({ 0, 1, 2 });
```

Comment:
Any way to verify that adding all actually is a no-op? Maybe take a vanilla test with no binding and add a variation with the full binding set and then verify the same schedule?

Reply:
I updated the test_synthetic_with_bindings_more_out() test to have its unconstrainted input instead list every output as a binding to ensure it still passes the same way.